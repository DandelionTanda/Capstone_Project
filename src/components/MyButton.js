import { View, Button, StyleSheet, TouchableOpacity, Text } from "react-native";
import React from "react";
import { MaterialCommunityIcons, Ionicons } from '@expo/vector-icons';
const MyButton = ({ onPress, title, buttonStyle, textStyle }) => {

  return (
    <TouchableOpacity onPress={onPress} style={buttonStyle}>
      {title==='Log out'?<Ionicons name="log-out" size={35} color="white" testID="log-out-icon"/>:null}
      <Text style={textStyle}>{title}</Text>
    </TouchableOpacity>
  );
};

export default MyButton;