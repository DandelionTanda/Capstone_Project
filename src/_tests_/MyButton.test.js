import MyButton from "../components/MyButton";
import renderer from 'react-test-renderer';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import React from 'react'
describe('My Button', () => {
  it('Should render correctly', () => {
    const { toJSON } = render(
      <MyButton title="Login"/>
    );
    expect(toJSON()).toMatchSnapshot();
  })
  it('Should display given title correctly', () => {
    const { getByText } = render(
      <MyButton title="Login"/>
    );
    expect(getByText('Login')).toBeTruthy();
  })
  it('Should display logout icon when title is log out', async () => {
    
    const { queryByTestId,getByText } = render(
      <MyButton title="Log out"/>
    );
    expect(getByText('Log out')).toBeTruthy();
    await waitFor(() => {
      expect(queryByTestId("log-out-icon")).toBeTruthy();
    })
  });
  it('Test click event', () => {
    const mockCallBack = jest.fn();
    const { getByText } = render(
      <MyButton onPress={mockCallBack} title="Login"/>
    );
    fireEvent.press(getByText("Login"))
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});


