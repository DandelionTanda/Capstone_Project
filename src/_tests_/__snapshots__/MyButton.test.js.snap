// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Button Should render correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text>
    Login
  </Text>
</View>
`;
